napari/utils/menus.py

```diff
@@ -0,0 +1,60 @@
+from typing import Callable, List, Optional
+
+from pydantic import Field
+
+from .events import EventedModel
+
+
+class MenuItem(EventedModel):
```

Comment:
haven't checked yet, but will these objects' events ever be connected to?  i.e. will we do `MenuItem().events.label.connect(...)`, or are we pretty much always going to be clearing and rebuilding with the `build_tools_menu` command?  If not, it might be better to simply inherit from `pydantic.BaseModel` rather than `EventedModel` until we know we need to monitor changes to a MenuItem instance

Reply:
in the future, I want Qt menus to be directly controlled by the Menu models defined here. however, currently, the only usage of the event system is when handling checkable items